pom.xml

```diff
@@ -111,7 +117,14 @@
             <version>4.12</version>
             <scope>test</scope>
         </dependency>
+        <dependency>
+            <groupId>org.checkerframework</groupId>
+            <artifactId>checker-qual</artifactId>
+            <version>2.4.0</version>
+            <optional>true</optional>
```

Comment:
I'd remove this - leave exclusions to consumers of this library.
https://github.com/google/guava/issues/2721

Reply:
Sure, you can try `<scope>provided</scope>`.

Reply:
Provided scope works - but means checker-qual has to be specified in each of the module poms.

It really seems like setting `optional` to true is the best solution here?

> Indicates the dependency is optional for use of this library. While the version of the dependency will be taken into account for dependency calculation if the library is used elsewhere, it will not be passed on transitively

The dependency is optional for the **use** of configurate, and doesn't need to be passed on transitively.